datadog/provider.go

```diff
@@ -125,6 +122,24 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("DD_HTTP_CLIENT_RETRY_TIMEOUT", nil),
 				Description: "The HTTP request retry timeout period. Defaults to 60 seconds.",
 			},
+			"http_client_retry_backoff_multiplier": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("DD_HTTP_CLIENT_RETRY_BACKOFF_MULTIPLIER", nil),
+				Description: "The HTTP request retry back off multiplier. Defaults to 2.",
+			},
+			"http_client_retry_backoff_base": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("DD_HTTP_CLIENT_RETRY_BACKOFF_BASE", nil),
+				Description: "The HTTP request retry back off base. Defaults to 2.",
+			},
+			"http_client_retry_max_retries": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("DD_HTTP_CLIENT_RETRY_MAX_RETRIES", nil),
+				Description: "The HTTP request maximum retry number. Defaults to 3.",
+			},
```

Comment:
Lets add some constraints around this (Min/Max) so end users cant configure the provider to continuously hit the api.

Reply:
I put max retry to 20 max , and backoff base and multiplier have to be at least 0. Feel free to let me know if you think other numbers are more adequate.

Reply:
For max retries, lets set the max to 5